source/js/plugins/pixel-wrapper.test.js

```diff
@@ -0,0 +1,197 @@
+    });
+
+    test('layer 1 activation toggle changes its class to deactivated', async () => {
+        let layer1Toggle = await browser.findElement(By.id('layer-1-activation'));
+        expect(await layer1Toggle.getAttribute('class')).toBe('layer-activated');
+
+        const actions = browser.actions();
+        await actions.click(layer1Toggle).perform();
+
+        expect(await layer1Toggle.getAttribute('class')).toBe('layer-deactivated');
+    });
+});
+
+// switch to Pixel.js with no input layers
+describe('Pixel Without Input', () => {
+    test('new URL get successfully loads other Pixel.js app', async () => {
+        await browser.get(urlNoInput);
+        let title = await browser.getTitle();
+        expect(title).toBe('Pixel.js');
+    });
+
+    test('activating plugin creates prompt for number of layers', async () => {
+        // activate plugin
+        let pluginIcon = await browser.findElement(By.id('diva-1-pixel-icon-glyph'));
+        const actions = browser.actions();
+        await actions.click(pluginIcon).perform();
+
+        // handle alert, create 3 layers by default
+        let alert = await browser.switchTo().alert();
+        expect(await alert.getText()).toContain("How many layers will you classify?");
+        await alert.accept();
+
+        // close tutorial
+        await browser.sleep(500);
+        let tutorialFooter = await browser.findElement(By.id('modal-footer'));
+        await actions.click(tutorialFooter).perform();
+    });
+
+    test('default 3 layers were created properly (layer 3 selector is visible)', async () => {
+        let layer3 = await browser.findElement(By.id('layer-3-selector'));
+        expect(await layer3.isDisplayed()).toBeTruthy();
+    });
+
+    test('tooltip creates tooltip-text on mouse hover', async () => {
+        let tooltip = await browser.findElement(By.className('tooltip'));
+        let tooltipText = await browser.findElement(By.className('tooltiptext'));
+
+        // should be hidden before hover
+        expect(await tooltipText.isDisplayed()).toBeFalsy();
+
+        // hover and should be visible
+        const actions = browser.actions();
+        await actions.click(tooltip).perform();
+        await browser.sleep(1000); // time for opacity transition
+        expect(await tooltipText.isDisplayed()).toBeTruthy();
+    });
+
+    test('deactivating and reactivating plugin no longer creates number of layers prompt', async () => {
+        // reactivate plugin
+        let pluginIcon = await browser.findElement(By.id('diva-1-pixel-icon-glyph'));
+        const actions = browser.actions();
+        await actions.click(pluginIcon).perform();
+        await actions.click(pluginIcon).perform();
+
+        // should go straight to tutorial
+        let tutorialDiv = await browser.findElement(By.id('tutorial-div'));
+        expect(await tutorialDiv.isDisplayed()).toBeTruthy();
+    });
+});
+
+
+
+
+
+
+
+
```

Comment:
Can you delete this whitespace? Careful not to commit this in the future.

Reply:
Done